tests/textcase/lsp/lsp_spec.lua

```diff
@@ -1 +1,62 @@
-describe("LSP", function() end)
+local test_helpers = require("tests.test_helpers")
+local textcase = require("textcase")
+
+local cur_dir = vim.fn.expand("%:p:h")
+
+---@param path string
+function M.read_file(path)
+  local fd = vim.loop.fs_open(path, "r", 438)
+  local fstat = vim.loop.fs_fstat(fd)
+  local contents = vim.loop.fs_read(fd, fstat.size, 0)
+  vim.loop.fs_close(fd)
+  return contents
+end
+
+function M.waitFor(max_time, callback)
+  local max_seconds = max_time
+  local curr_seconds = 0
+  local task_finished = false
+  while curr_seconds < max_seconds and not task_finished do
+    curr_seconds = curr_seconds + 0.5
+    task_finished = callback()
+    vim.wait(500, function() end)
+  end
+end
+
+describe("LSP", function()
+  describe("Rename", function()
+    before_each(function()
+      textcase.setup({})
+    end)
+
+    it("Should be triggered on keybinding", function()
+      local path = cur_dir .. "/tests/textcase/lsp/fixtures/component-camel-case.tsx"
+      local cmd = " silent exe 'e " .. path .. "'"
+      vim.cmd(cmd)
+      vim.bo.filetype = "typescriptreact"
+      -- allow tsserver start
+
+      local ts_server_started = false
+      M.waitFor(30, function()
+        vim.cmd("LspInfo")
```

Comment:
Cool idea to check for `LspInfo` :) I was already wondering how you would do that.

Reply:
Hopefully it does not become a flaky test. But we do depend on LspInfo Screen

What I'm trying to learn is not to be so perfectionist, but keep progressing and change things later if we find a better way

Reply:
There is `vim.lsp.buf.server_ready()` but it takes way longer.